gammapy/catalog/core.py

```diff
@@ -257,18 +258,44 @@ def _make_source_object(self, index):
             name_extended = data["Source_Name"].strip()
         else:
             name_extended = None
+
+        if self._source_name_key in data and self._source_name_key == "PSR_Name":
+            name_spectral = data[self._source_name_key].strip()
+        elif self._source_name_key in data and self._source_name_key == "PSRJ":
+            name_spectral = f"PSR{data[self._source_name_key].strip()}"
+        elif "Source_name" in data:
+            name_spectral = data["Source_name"].strip()
+        else:
+            name_spectral = None
```

Comment:
maybe move to a private function _get_name_spectral that you rewrite in the sub-classes corresponding to different cases